snorkel/learning/utils.py

```diff
@@ -1,14 +1,86 @@
 import math
+import matplotlib
 import matplotlib.pyplot as plt
 import numpy as np
 import scipy.sparse as sparse
+import warnings
+
+from pandas import DataFrame
 
-import matplotlib
 matplotlib.use('Agg')
-import warnings
 warnings.filterwarnings("ignore", module="matplotlib")
 
 
+class Scorer(object):
+    """Abstract type for scorers"""
+    def __init__(self, test_candidates, test_labels, gold_candidate_set):
+        """
+        test_candidates: candidates corresponding to test_labels
+        test_labels: ground truth labels for test candidates
+        gold_candidate_set: full set of gold labeled candidates
```

Comment:
I think this has been a point of confusion for people, so let's specify the exact types here?  So:

- List of `Candidate`s
- `csrLabelMatrix`
- `CandidateSet`

Right?